src/main/java/no/unit/nva/bare/UpdateAuthorityHandler.java

```diff
@@ -0,0 +1,151 @@
+package no.unit.nva.bare;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Handler for requests to Lambda function.
+ */
+public class UpdateAuthorityHandler implements RequestHandler<Map<String, Object>, GatewayResponse> {
+
+    public static final String AUTHORITY_NOT_FOUND = "Authority not found for 'scn = %s'";
+    public static final String TO_MANY_AUTHORITIES_FOUND = "To many authorities found for 'scn = %s'";
+    public static final String BODY_ARGS_MISSING = "Nothing to update. 'feideId' and 'orcId' are missing.";
+    public static final String MISSING_BODY_ELEMENT_EVENT = "Missing body element 'event'.";
+    public static final String MISSING_PATH_PARAMETER_SCN = "Missing path parameter 'scn'.";
+    public static final String COMMUNICATION_ERROR_WHILE_UPDATING = "Communication failure while updating authority %s";
+    public static final String SCN_KEY = "scn";
+    public static final String FEIDEID_KEY = "feideId";
+    public static final String ORCID_KEY = "orcId";
+    public static final String BODY_KEY = "body";
+    protected final transient GatewayResponse gatewayResponse = new GatewayResponse();
+    protected final transient AuthorityConverter authorityConverter = new AuthorityConverter();
+    protected final transient BareConnection bareConnection;
+
+
+    public UpdateAuthorityHandler() {
+        this.bareConnection = new BareConnection();
+    }
+
+    public UpdateAuthorityHandler(BareConnection bareConnection) {
+        this.bareConnection = bareConnection;
+    }
+
+    /**
+     * Main lambda function to update feideId or/and orcId on Bare authority metadata.
+     * @param input payload with body-parameter containing the authority metadata
+     * @return a GatewayResponse
+     */
+    @Override
+    @SuppressWarnings("unchecked")
+    public GatewayResponse handleRequest(final Map<String, Object> input, Context context) {
+        String bodyEvent = (String) input.get("body");
```

Comment:
Hm, litt flere try-catch for å unngå at exceptions kan lekke ut?

Reply:
Jeg ser hva du mener. Det er ikke fint med all disse "else-ifs", men jeg ser ikke helt hvordan koden kan bli penere da. Sorry.